actors/miner/tests/prove_replica_failures_test.rs

```diff
@@ -0,0 +1,208 @@
+        param_twiddle: Some(Box::new(|p: &mut ProveReplicaUpdates2Params| {
+            p.sector_proofs = vec![];
+            p.aggregate_proof = RawBytes::default();
+        })),
+        ..Default::default()
+    };
+    expect_abort_contains_message(
+        ExitCode::USR_ILLEGAL_ARGUMENT,
+        "exactly one of sector proofs or aggregate proof must be non-empty",
+        h.prove_replica_updates2_batch(&rt, &sector_updates, false, false, cfg),
+    );
+}
+
+#[test]
+fn reject_both_proof_types() {
+    let (h, rt, sector_updates) = setup(1, 0, 0, 0);
+    let cfg = ProveReplicaUpdatesConfig {
+        param_twiddle: Some(Box::new(|p: &mut ProveReplicaUpdates2Params| {
+            p.sector_proofs = vec![RawBytes::new(vec![1, 2, 3, 4])];
+            p.aggregate_proof = RawBytes::new(vec![1, 2, 3, 4])
+        })),
+        ..Default::default()
+    };
+    expect_abort_contains_message(
+        ExitCode::USR_ILLEGAL_ARGUMENT,
+        "exactly one of sector proofs or aggregate proof must be non-empty",
+        h.prove_replica_updates2_batch(&rt, &sector_updates, false, false, cfg),
+    );
+}
+
+#[test]
+fn reject_mismatched_proof_len() {
+    let (h, rt, sector_updates) = setup(1, 0, 0, 0);
+    let cfg = ProveReplicaUpdatesConfig {
+        param_twiddle: Some(Box::new(|p: &mut ProveReplicaUpdates2Params| {
+            p.sector_proofs.push(RawBytes::new(vec![1, 2, 3, 4]));
+        })),
+        ..Default::default()
+    };
+    expect_abort_contains_message(
+        ExitCode::USR_ILLEGAL_ARGUMENT,
+        "mismatched lengths",
+        h.prove_replica_updates2_batch(&rt, &sector_updates, false, false, cfg),
+    );
+}
+
+#[test]
+fn reject_aggregate_proof() {
+    let (h, rt, sector_updates) = setup(1, 0, 0, 0);
+    let cfg = ProveReplicaUpdatesConfig {
+        param_twiddle: Some(Box::new(|p: &mut ProveReplicaUpdates2Params| {
+            p.sector_proofs = vec![];
+            p.aggregate_proof = RawBytes::new(vec![1, 2, 3, 4])
+        })),
+        ..Default::default()
+    };
+    expect_abort_contains_message(
+        ExitCode::USR_ILLEGAL_ARGUMENT,
+        "aggregate update proofs not yet supported",
+        h.prove_replica_updates2_batch(&rt, &sector_updates, false, false, cfg),
+    );
+}
+
+#[test]
+fn reject_all_proofs_fail() {
+    let (h, rt, sector_updates) = setup(2, 0, 0, 0);
+    let cfg = ProveReplicaUpdatesConfig { proofs_valid: vec![false, false], ..Default::default() };
+    expect_abort_contains_message(
+        ExitCode::USR_ILLEGAL_ARGUMENT,
+        "no valid updates",
+        h.prove_replica_updates2_batch(&rt, &sector_updates, false, false, cfg),
+    );
+}
+
+#[test]
+fn reject_invalid_update() {
+    let (h, rt, mut sector_updates) = setup(2, 0, 0, 0);
+    sector_updates[1].sector = sector_updates[0].sector;
+    let cfg = ProveReplicaUpdatesConfig::default();
+    expect_abort_contains_message(
+        ExitCode::USR_ILLEGAL_ARGUMENT,
+        "aggregate update proofs not yet supported",
```

Comment:
Can you leave a comment explaining this somewhat counter-intuitive error message?

Reply:
Oops, that was the test that's failing because the expected message is wrong! Fixed the expectation.